src/tools/crossgen2/ILCompiler.ReadyToRun/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -118,9 +118,20 @@ private void EmitPInvokeCall(PInvokeILCodeStreams ilCodeStreams)
 
         private MethodIL EmitIL()
         {
+            if (!_importMetadata.Flags.PreserveSig)
+                throw new NotSupportedException();
+
             PInvokeILCodeStreams pInvokeILCodeStreams = new PInvokeILCodeStreams();
             ILEmitter emitter = pInvokeILCodeStreams.Emitter;
+            ILCodeStream marshallingCodestream = pInvokeILCodeStreams.MarshallingCodeStream;
             ILCodeStream unmarshallingCodestream = pInvokeILCodeStreams.UnmarshallingCodestream;
+            ILCodeStream cleanupCodestream = pInvokeILCodeStreams.CleanupCodeStream;
+
+            // Marshalling is wrapped in a finally block to guarantee cleanup
+            ILExceptionRegionBuilder tryFinally = emitter.NewFinallyRegion();
```

Comment:
This relies on the codegen to remove empty finally regions. I assume it's fine since RyuJIT does that. I kind of forgot to design the ILExceptionRegionBuilder in a way where we could change our mind about the exception region later...